gdal/frmts/gtiff/libtiff/tif_webp.c

```diff
@@ -143,28 +143,27 @@ TWebPDecode(TIFF* tif, uint8* op, tmsize_t occ, uint16 s)
     }
     return 0;
   } else {
-    int current_y, width, stride;
+    int current_y, stride;
     uint8_t* buf;
-    
+
     // Returns the RGB/A image decoded so far
-    buf = WebPIDecGetRGB(sp->psDecoder, &current_y, &width, NULL, &stride);
-    
-    if ((current_y - sp->last_y) * width * stride < occ) {
+    buf = WebPIDecGetRGB(sp->psDecoder, &current_y, NULL, NULL, &stride);
+
+    if ((current_y - sp->last_y) * stride > occ) {
       TIFFErrorExt(tif->tif_clientdata, module,
           "Not enough data at scanline %d (short %lu bytes)",
           current_y,
-          occ - ((current_y - sp->last_y) * width * stride));
+          occ - ((current_y - sp->last_y) * stride));
       return 0;
     }
 
     memcpy(op,   
-       buf + (sp->last_y * stride * width),
+       buf + (sp->last_y - 1) * stride,
```

Comment:
I'm having a hard time with the convention of last_y being initialized to 1, and then here 1 being substracted. All this boils down to the actual meaning of current_y in WebPIDecGetRGB ? https://github.com/webmproject/libwebp/blob/master/src/webp/decode.h#L338 says that "*last_y is the index of last decoded row". So if the first line is the last decoded, shouldn't current_y be 0 ? (but if the line can only be partially decoded, what WebPIDecGetRGB() would return ?) In which case, I'm not sure if current_y - last_y in the test above wouldn't be evaluated to a huge unsigned value and error out ?

Reply:
me to! I compressed a tiff by strip (1 row per strip) and current_y was set to 1 for the first row when decoding which goes against convention.

Is is possible to have a partial row passed to decode?

Is it possible to have more than one row (when encoding with strips) passed to decode?

I will dig into the source some more but if you know then it will save me some time.

Reply:
> Is is possible to have a partial row passed to decode?

Normally not something people would do (GDAL won't), but the API allows for it. Some codecs that don't support partial row docoding actually error out if occ is not a multiple of the scanline size. See Fax3Decode2D() in tif_fax3.c for example. Looks like we would need that too.

> Is it possible to have more than one row (when encoding with strips) passed to decode?

Yes, that would be something possible (but not triggered through GDAL use of libtiff API)

Reply:
I am going to make `last_y` start from zero to stick with convention. `last_y` calculates the offset in the buffer so it is possible to ignore that `WebPIDecGetRGB` indexes from 1.